File: frontend/src/types/budget.ts
// 예산 관리 타입 정의

export enum BudgetStatus {
  PLANNED = 'PLANNED',      // 계획됨
  IN_PROGRESS = 'IN_PROGRESS',  // 진행 중
  COMPLETED = 'COMPLETED',    // 완료
  OVERBUDGET = 'OVERBUDGET',   // 예산 초과
  CANCELLED = 'CANCELLED'     // 취소
}

export enum Priority {
  HIGH = 'HIGH',    // 높음
  MEDIUM = 'MEDIUM',  // 보통
  LOW = 'LOW'      // 낮음
}

export const BudgetStatusLabels = {
  [BudgetStatus.PLANNED]: '계획됨',
  [BudgetStatus.IN_PROGRESS]: '진행 중',
  [BudgetStatus.COMPLETED]: '완료',
  [BudgetStatus.OVERBUDGET]: '예산 초과',
  [BudgetStatus.CANCELLED]: '취소'
};

export const PriorityLabels = {
  [Priority.HIGH]: '높음',
  [Priority.MEDIUM]: '보통',
  [Priority.LOW]: '낮음'
};

export interface Budget {
  id: number;
  category: string;
  itemName: string;
  plannedAmount: number;
  actualAmount?: number;
  status: BudgetStatus;
  priority: Priority;
  description?: string;
  vendor?: string;
  dueDate?: string;
  createdAt: string;
  updatedAt: string;
  isOverBudget: boolean;
  remainingBudget: number;
}

export interface BudgetFormData {
  category: string;
  itemName: string;
  plannedAmount: string;
  actualAmount: string;
  status: BudgetStatus;
  priority: Priority;
  description: string;
  vendor: string;
  dueDate: string;
}

export interface BudgetSummary {
  category: string;
  itemCount: number;
  totalPlannedAmount: number;
  totalActualAmount: number;
  remainingAmount: number;
  completionRate: number;
}

// 예산 카테고리 기본값
export const DEFAULT_BUDGET_CATEGORIES = [
  '웨딩홀',
  '스튜디오',
  '드레스',
  '메이크업',
  '부케/플라워',
  '초대장',
  '답례품',
  '혼수',
  '신혼여행',
  '기타'
];

File: src/main/java/com/w/p/entity/Budget.java
package com.w.p.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "WP_BUDGETS")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Budget {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private Long userId;
    
    @Column(nullable = false, length = 100)
    private String category;
    
    @Column(nullable = false, length = 200)
    private String itemName;
    
    @Column(precision = 15, scale = 2)
    private BigDecimal plannedAmount;
    
    @Column(precision = 15, scale = 2)
    private BigDecimal actualAmount;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private BudgetStatus status;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Priority priority;
    
    @Column(length = 500)
    private String description;
    
    @Column(length = 200)
    private String vendor;
    
    private LocalDateTime dueDate;
    
    @CreationTimestamp
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    private LocalDateTime updatedAt;
    
    public enum BudgetStatus {
        PLANNED,      // 계획됨
        IN_PROGRESS,  // 진행 중
        COMPLETED,    // 완료
        OVERBUDGET,   // 예산 초과
        CANCELLED,    // 취소
        APPROVED,     // 승인됨
        REJECTED      // 반려됨
    }
    
    public enum Priority {
        HIGH,    // 높음
        MEDIUM,  // 보통
        LOW      // 낮음
    }
    
    // 예산 초과 여부 확인
    public boolean isOverBudget() {
        if (actualAmount == null || plannedAmount == null) {
            return false;
        }
        return actualAmount.compareTo(plannedAmount) > 0;
    }
    
    // 잔여 예산 계산
    public BigDecimal getRemainingBudget() {
        if (actualAmount == null || plannedAmount == null) {
            return plannedAmount != null ? plannedAmount : BigDecimal.ZERO;
        }
        return plannedAmount.subtract(actualAmount);
    }
}
